util/gen_action_info.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python
+
+import csv
+import json
+import os
+import urllib.request
+import re
+import io
+from collections import defaultdict
+
+"""Action data is available in csv form that has 3 headers looks like:
+key,0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65
+#,Name,,Icon,ActionCategory,,Animation{Start},VFX,Animation{End},ActionTimeline{Hit},,ClassJob,BehaviourType,ClassJobLevel,IsRoleAction,Range,CanTargetSelf,CanTargetParty,CanTargetFriendly,CanTargetHostile,,,TargetArea,,,,CanTargetDead,,CastType,EffectRange,XAxisModifier,,PrimaryCost{Type},PrimaryCost{Value},SecondaryCost{Type},SecondaryCost{Value},Action{Combo},PreservesCombo,Cast<100ms>,Recast<100ms>,CooldownGroup,,MaxCharges,AttackType,Aspect,ActionProcStatus,,Status{GainSelf},UnlockLink,ClassJobCategory,,,AffectsPosition,Omen,IsPvP,,,,,,,,,,,,IsPlayerAction
+int32,str,bit&01,Image,ActionCategory,byte,ActionCastTimeline,ActionCastVFX,ActionTimeline,ActionTimeline,byte,ClassJob,byte,byte,bit&02,sbyte,bit&04,bit&08,bit&10,bit&20,bit&40,bit&80,bit&01,bit&02,bit&04,sbyte,bit&08,bit&10,byte,byte,byte,bit&20,byte,uint16,byte,Row,Action,bit&40,uint16,uint16,byte,byte,byte,AttackType,byte,ActionProcStatus,byte,Status,Row,ClassJobCategory,byte,bit&80,bit&01,Omen,bit&02,bit&04,bit&08,bit&10,bit&20,bit&40,bit&80,bit&01,bit&02,byte,bit&04,bit&08,bit&10
+0,"",False,405,0,0,0,0,0,0,0,-1,0,0,False,0,False,False,False,False,False,False,False,False,False,0,True,True,0,0,0,False,0,0,0,0,0,False,0,0,0,0,0,0,0,0,0,0,0,0,0,False,False,0,False,False,True,False,True,True,False,False,True,0,False,False,False
+===rest of data===
+
+We take the raw output and convert it into a usable JavaScript file to be imported and used within modules so we don't need to find and hardcode action data.
+
+TODO:
+Move constants into config file
+Implement argparse
+Make more universal functions more friendly for reuse
+General streamlining
+Implement column filtering on returned data
+Validate output
+Implement proper logging
+Implement proper error handling
+Review and streamline name normalization (current normalization taken from csv_util.py and coinach.py)
+Build out local file functionality
+Move table structuring from dunders to get_data_table()
+Stab feature creep/overengineering in the eye
+"""
+
+config = {
+    "output": {
+        "pve": "pve_action_info.js",
+        "pvp": "pvp_action_info.js",
+        "crafting": "crafting_action_info.js",
+        "combo": "pve_action_combos.js",
```

Comment:
I think if these don't exist yet, it'd be better to not include them until they do.

Reply:
Agreed.  I welcome suggestions on how I could show that something is possible even when it isn't immediately applicable.  I thought this script out more of a module, not really as a script specifically for this single purpose.  I realize a lot of these things don't make sense here, but I didn't want to artificially limit ideas from others.

Reply:
I think a better way to do this would be a `# TODO: handle other action types here in the future` rather than adding code for things that don't exist.